indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/pki/X509Certificate.kt

```diff
@@ -282,7 +312,38 @@ data class X509Certificate @Throws(IllegalArgumentException::class) constructor(
 
     val publicKey: CryptoPublicKey get() = tbsCertificate.publicKey
 
-    companion object : PemDecodable<Asn1Sequence, X509Certificate>(EB_STRINGS.DEFAULT, EB_STRINGS.LEGACY) {
+    val criticalExtensionOids: MutableSet<ObjectIdentifier>
+        get() = this.tbsCertificate.extensions
+            ?.filter { it.critical }
+            ?.mapTo(mutableSetOf()) { it.oid }
+            ?: mutableSetOf()
```

Comment:
X509Certificate instances should be immutable otherwise you can end up with inconsistent state. For validation you can use `toMutableSet()` and safely work on your internal copy instead.